plugins/uartdmx/UartDmxThread.cpp

```diff
@@ -110,7 +115,31 @@ void *UartDmxThread::Run() {
 
   framesleep:
     // Sleep for the remainder of the DMX frame time
-    usleep(m_malft);
+    clock.CurrentMonotonicTime(&ts2);
+    TimeInterval elapsed = ts2 - ts1;
+
+    if (m_granularity == GOOD) {
+      while (elapsed.InMilliSeconds() < frameTime) {
+        usleep(1000);
+        clock.CurrentMonotonicTime(&ts2);
+        elapsed = ts2 - ts1;
+      }
+    } else {
+      // See if we can drop out of bad mode.
+      usleep(1000);
+      clock.CurrentMonotonicTime(&ts3);
+      TimeInterval interval = ts3 - ts2;
+      if (interval.InMilliSeconds() <= BAD_GRANULARITY_LIMIT) {
+        m_granularity = GOOD;
+        OLA_INFO << "Switching from BAD to GOOD granularity for UART thread";
+      }
+      
+      elapsed = ts3 - ts1;
```

Comment:
/plugins/uartdmx/UartDmxThread.cpp:136:  Line ends in whitespace.  Consider deleting these extra spaces. [whitespace/end_of_line] [4]
```suggestion
      }

      elapsed = ts3 - ts1;
```